app/models/mixins/relationship_mixin.rb

```diff
@@ -673,11 +673,7 @@ def puts_relationship_tree
   def self.deprecate_of_type_parameter(*args)
     return args if args.empty? || args.first.kind_of?(Hash)
 
-    unless Rails.env.production?
-      msg = "[DEPRECATION] of_type parameter without hash symbol is deprecated.  Please use :of_type => 'Type' style instead.  At #{caller[1]}"
-      $log.warn msg
-      warn msg
-    end
+    Vmdb::Deprecation.deprecation_warning("of_type parameter without hash symbol", "use :of_type => 'Type' style instead")
```

Comment:
It sounds like this and a few others require the inline conditional since they are only internal API, customers would not be using them and we don't want to flood the log with messages.